frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -332,6 +333,85 @@ pub mod pallet {
+            log::info!(
+                "PreApprovedDevelopers: {}",
+                current_pre_approved_developers_count
+            );
+
+            assert_eq!(Version::V3_0_0, StorageVersion::<T>::get());
+
+            Ok(())
+        }
+
+        fn on_runtime_upgrade() -> Weight {
+            // Maintenance mode to prevent existing extrinsics update PreApprovedDevelopers map.
+            PalletDisabled::<T>::put(true);
+            StorageVersion::<T>::put(Version::default());
+            let consumed_weight = cleanup_pre_approved_developers::<T>();
+            PalletDisabled::<T>::put(false);
+            consumed_weight
+        }
+
+        #[cfg(feature = "try-runtime")]
+        fn post_upgrade() -> Result<(), &'static str> {
+            log::info!(">>> Post Upgrade");
+
+            assert_eq!(Version::V4_0_0, StorageVersion::<T>::get());
+            assert_eq!(PalletDisabled::<T>::get(), false);
+
+            // pre approved developers count should be 0
+            let current_pre_approved_developers_count =
+                PreApprovedDevelopers::<T>::iter_keys().count() as u64;
+            log::info!(
+                "PreApprovedDevelopers: {}",
+                current_pre_approved_developers_count
+            );
+            assert_eq!(current_pre_approved_developers_count, 0);
+
+            Ok(())
+        }
+    }
+
+    fn cleanup_pre_approved_developers<T: Config>() -> Weight {
+        let mut consumed_weight = Zero::zero();
+
+        let deletion_weight = T::DbWeight::get().writes(1) * 11 / 10;
+        let approximate_deletions_remaining =
+            (T::BlockWeights::get().max_block / (T::DbWeight::get().writes(1) * 11 / 10)).max(1);
+
+        log::info!(
+            "Approximate Delegations Remaining {}",
+            approximate_deletions_remaining
+        );
+
+        let removal_result = if cfg!(feature = "try-runtime") {
+            PreApprovedDevelopers::<T>::remove_all(None)
+        } else {
+            PreApprovedDevelopers::<T>::remove_all(Some(approximate_deletions_remaining as u32))
+        };
+
+        match removal_result {
+            KillStorageResult::AllRemoved(removed_entries_num) => {
+                log::info!(">>> PreApprovalDevelopers cleanup finished.");
+                consumed_weight += deletion_weight * removed_entries_num as u64;
+                consumed_weight
+            }
+            KillStorageResult::SomeRemaining(removed_entries_num) => {
+                log::info!(">>> PreApprovalDevelopers cleanup stopped due to reaching max amount of deletions.");
+                consumed_weight += deletion_weight * removed_entries_num as u64;
+                if cfg!(feature = "try-runtime") {
+                    return consumed_weight + cleanup_pre_approved_developers::<T>();
+                } else {
+                    return consumed_weight;
+                }
+            }
+        }
     }
```

Comment:
This isn't needed since this cleanup is lightweight.

Just add `PreApprovedDevelopers::<T>::remove_all(None)` to the original call and it will be fine.

Reply:
Practically, this is good because they are very lightweight?
```
fn on_runtime_upgrade() -> Weight {
            // lightweight deletions
            StorageVersion::<T>::put(Version::default());
            PreApprovalIsEnabled::<T>::kill();
            PreApprovedDevelopers::<T>::remove_all(None);
            0
 }
```

Reply:
Yes, it's relatively lightweight but that code snippet is incorrect.

You should still return the consumed weight.

1 write + 1 write + `N` writes.
`remove_all` will return an enum containing the amount of removed entries so you can use that.

Reply:
I understand what you meant. thank you!